test/terra/reference/ref_snapshot_amplitudes.py

```diff
@@ -0,0 +1,110 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Test circuits and reference outputs for snapshot amplitude instructions.
+"""
+
+from numpy import array, sqrt
+from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
+from qiskit.providers.aer.extensions.snapshot import Snapshot
+from qiskit.providers.aer.extensions.snapshot_amplitudes import *
+from qiskit.providers.aer.extensions.snapshot_statevector import *
+
+def snapshot_amplitudes_labels_params():
+    """Dictionary of labels and params for 3-qubit amplitude snapshots"""
+    return {
+        "[0]": [0],
+        "[7]": [7],
+        "[0,1]": [0,1],
+        "[7,3,5,1]": [7,3,5,1],
+        "[4,1,5]": [4,1,5],
+        "[6,2]": [6,2],
+        "all": [0,1,2,3,4,5,6,7]
```

Comment:
Do you test hexa?

Reply:
Right, added a couple of tests for hexa.